lib/model/classycat_schema_create.py

```diff
@@ -0,0 +1,170 @@
+            "Classify the following items:\n<INPUT>\n<INSERT_ITEMS_HERE>\n</INPUT>"
+        )
+
+
+    def create_schema(self, schema_name, topics, examples, languages):
+        schema_id = str(uuid.uuid4())
+        schema_filename = f"{schema_id}.json"
+        schema_id_mapping_filename = f"{schema_name}.json"
+
+        schema = {
+            "schema_id": schema_id,
+            "schema_name": schema_name,
+            "topics": topics,
+            "examples": examples,
+            "languages": languages,
+        }
+
+        schema["prompt"] = self.generate_prompt_from_schema(schema)
+
+        upload_file_to_s3(self.output_bucket, schema_filename, json.dumps(schema))
+        # s3_client.put_object(
+        #     Bucket=output_bucket, Key=schema_filename, Body=json.dumps(schema)
+        # )
+
+        upload_file_to_s3(self.output_bucket, schema_id_mapping_filename, json.dumps({"schema_id": schema_id}))
+        # s3_client.put_object(
+        #     Bucket=output_bucket,
+        #     Key=schema_id_mapping_filename,
+        #     Body=json.dumps({"schema_id": schema_id}),
+        # )
+        logger.info(f"Schema {schema_name} created with id {schema_id}")
+
+        return schema_id
+
+
+    def generate_prompt_from_schema(self, schema):
+        topics = schema["topics"]
+        examples = schema["examples"]
+        languages = (
+            " and ".join(schema["languages"])
+            if len(schema["languages"]) <= 2
+            else ", ".join(schema["languages"][:-1]) + f", and {schema['languages'][-1]}"
+        )
+        prompt_categories = [
+            "<TAXONOMY>",
+            f"<TITLE>{schema['schema_name']}</TITLE>",
+            "<TOPICS>",
+        ]
+
+        for i, topic in enumerate(topics):
+            prompt_categories.append(
+                f"<TOPIC_{i}><TITLE>{topic['topic']}</TITLE><DESCRIPTION>{topic['description']}</DESCRIPTION></TOPIC_{i}>"
+            )
+
+        prompt_categories.append("</TOPICS>")
+
+        prompt_categories.append("<EXAMPLES>")
+        prompt_categories.append("<INPUT>")
+
+        for i, example in enumerate(examples):
+            prompt_categories.append(f"<ITEM_{i}>{example['text']}</ITEM_{i}>")
+
+        prompt_categories.append("</INPUT>")
+        prompt_categories.append("<OUTPUT>")
+
+        for i, example in enumerate(examples):
+            prompt_categories.append(
+                f"<CATEGORIES_{i}>{';'.join(example['labels'])}</CATEGORIES_{i}>"
+            )
+
+        prompt_categories.append("</OUTPUT>")
+        prompt_categories.append("</EXAMPLES>")
+
+        prompt_categories.append("</TAXONOMY>")
+
+        return self.base_prompt.format(
+            categories="\n".join(prompt_categories),
+            languages=languages,
+            schema_name=schema["schema_name"],
+        )
+
+
+    def process(self, message: Message) -> ClassyCatSchemaResponse:
+        # unpack parameters for create_schema
```

Comment:
having an example of the schema response in comments here would be really helpful